### PR TITLE
Replace inline snippets with external compilable snippets

### PR DESCRIPTION
**Pull request by zoewangg**
## Motivation and Context
Replace inline snippets with external snippets so that we can make sure sample code compiles

## Modifications
Use `@snippet` introduced in Java 18 in S3 transfer manager for code snippets 
https://docs.oracle.com/en/java/javase/18/code-snippet/index.html#external-snippets

## Testing
Tested it locally and verified the generated Javadoc, will test it using CodeBuild

### CODE REVIEW

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java [hunk 1]
```diff
@@ -54,58 +54,27 @@
...
- * 
- * // Attach a TransferListener
- * FileUpload upload =
- *     tm.uploadFile(u -> u.source(Paths.get("myFile.txt"))
- *                         .putObjectRequest(p -> p.bucket("bucket").key("key"))
- *                         .addTransferListener(LoggingTransferListener.create()));
- * upload.completionFuture().join();
- * }
- * </pre>
+ * <h1>Instantiate a Transfer Manager</h1>
```
Comment (millems) [inline services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java:62]: Nit: Instantiate Transfer Manager, or Create a transfer manager instance

File: services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java [hunk 2]
```diff
@@ -54,58 +54,27 @@
...
+ * <b>Create a transfer manager with SDK default settings</b>
```
Comment (millems) [inline services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/S3TransferManager.java:63]: Nit: Create a transfer manager instance

File: services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/samples/S3TransferManagerSamples.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        // @start region=resumeDownloadFile
+        S3TransferManager transferManager = S3TransferManager.create();
+
+        // Initiate the transfer
+        FileDownload download =
+            transferManager.downloadFile(d -> d.getObjectRequest(g -> g.bucket("bucket").key("key")).destination(Paths.get(
+                "myFile.txt")));
+
+        // Pause the download
+        ResumableFileDownload resumableFileDownload = download.pause(); // @link substring="pause" target="software.amazon.awssdk.transfer.s3.model.FileDownload#pause()"
```
Comment (millems) [inline services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/samples/S3TransferManagerSamples.java:110]: @link seems cool. Where should we be using it?

Comment (zoewangg) [reply services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/samples/S3TransferManagerSamples.java:110]: Yup. We can use it for any method we want to link to its Javadoc. The link is not super visible from the javadoc site though
Screen Shot 2022-10-06 at 1 27 13 PM

File: pom.xml [hunk 4]
```diff
@@ -710,6 +713,17 @@
             </properties>
         </profile>
 
+        <profile>
+            <id>jdk-18-plus</id>
```
Comment (millems) [inline pom.xml:717]: Does our Javadoc publish job use JDK 18+?

**Review by millems — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (zoewangg) [reply pom.xml:717]: Yes! 
https://github.com/aws/aws-sdk-java-v2/blob/92e48e4e62ad723f2e94b6fcbcce0238c35a731c/buildspecs/release-javadoc.yml#L4

**Pull request merged**